server/controllers/refer/courseParticipationsController.ts

```diff
@@ -104,12 +143,58 @@ export default class CourseParticipationsController {
 
       FormUtils.setFieldErrors(req, res, ['courseId', 'otherCourseName'])
 
-      res.render('referrals/courseParticipations/new', {
+      res.render('referrals/courseParticipations/course', {
+        action: referPaths.programmeHistory.create({ referralId: referral.id }),
         courseRadioOptions: CourseUtils.courseRadioOptions(courses),
+        otherCourseNameChecked: !!res.locals.errors.messages.otherCourseName,
         pageHeading: 'Add Accredited Programme history',
         person,
         referralId: referral.id,
+        values: {},
+      })
+    }
+  }
+
+  updateCourse(): TypedRequestHandler<Request, Response> {
+    return async (req: Request, res: Response) => {
+      TypeUtils.assertHasUser(req)
+
+      const currentCourseParticipation = await this.courseService.getParticipation(
+        req.user.token,
+        req.params.courseParticipationId,
+      )
+
+      if (!currentCourseParticipation) {
```

Comment:
This should already be handled in the service.

Reply:
Ah, I thought you'd handled it somewhere!

Reply:
I've updated the `getParticipation` return type [here](https://github.com/ministryofjustice/hmpps-accredited-programmes-ui/pull/245/commits/1ed0e84196e9fe7294f1ff5ff9290fe1dcbe41ed) - hope that's correct!